tests/test_meta/test_hierarchical_predictor.py

```diff
@@ -38,14 +42,18 @@ def make_hierarchical_dataset(task):
     else:
         raise ValueError("Invalid task")
 
-    X = pd.DataFrame(X, columns=[f"x_{i}" for i in range(X.shape[1])]).assign(
-        g_0=1,
-        g_1=["A"] * (n_samples // 2) + ["B"] * (n_samples // 2),
-        g_2=["X"] * (n_samples // 4) + ["Y"] * (n_samples // 2) + ["Z"] * (n_samples // 4),
+    X_ = (
+        pd.DataFrame(X, columns=[f"x_{i}" for i in range(X.shape[1])])
+        .assign(
+            g_0=1,
+            g_1=["A"] * (n_samples // 2) + ["B"] * (n_samples // 2),
+            g_2=["X"] * (n_samples // 4) + ["Y"] * (n_samples // 2) + ["Z"] * (n_samples // 4),
+        )
+        .pipe(frame_func)
```

Comment:
@MarcoGorelli here is where I encountered the [`with_columns` issue](https://github.com/narwhals-dev/narwhals/issues/141). My first intention was to do it as:

```py
    (
     frame_func(dict(zip([f"x_{i}" for i in range(X.shape[1])]), X.T)
     .pipe(nw.from_native)
     .with_columns(
            g_0=1,
            g_1=["A"] * (n_samples // 2) + ["B"] * (n_samples // 2),
            g_2=["X"] * (n_samples // 4) + ["Y"] * (n_samples // 2) + ["Z"] * (n_samples // 4),
        )
        .pipe(nw.to_native)
```

Reply:
I think Polars doesn't do what you're expecting it to here 😉 
```python
In [34]: frame = pl.DataFrame({"a": [1,2,3]})

In [35]: frame.with_columns(b=['a', 'b', 'c'])
Out[35]:
shape: (3, 2)
┌─────┬─────────────────┐
│ a   ┆ b               │
│ --- ┆ ---             │
│ i64 ┆ list[str]       │
╞═════╪═════════════════╡
│ 1   ┆ ["a", "b", "c"] │
│ 2   ┆ ["a", "b", "c"] │
│ 3   ┆ ["a", "b", "c"] │
└─────┴─────────────────┘
```

I _think_ what we need is a Series constructor. Something like
```python
df.with_columns(
    g_1=nw.Series(
        ["A"] * (n_samples // 2) + ["B"] * (n_samples // 2),
        implementation=nw.get_implementation(X),
    )
)
```
?

Reply:
Maybe we should move this conversation to the [issue](https://github.com/narwhals-dev/narwhals/issues/141) 😇

> I think Polars doesn't do what you're expecting it to here 😉

You are right, but it does with numpy.

> We need to pass something to nw.Series, otherwise it doesn't know which library's Series to create

Doesn't `with_columns` has some context (from self)? (As mentioned on Discord I need to catch up a lot with the internals, black magic to me)

Reply:
>  it does with numpy.

Just pushed out a release which can handle numpy 😎 It may take a few minutes to appear

> Wouldn't with_columns has some context (from self)?

Yeah, I'm just trying to figure out how to stay compatible with the Polars API. Because if you want to create a new column with `[1, 2, 3]` in Polars, you'd do
```
df.with_columns(d=pl.Series([1,2,3]))
```
So I think we need to instruct the user to construct a `nw.Series`, in such a way that the series can exist in a self-standing way without the `df.with_columns` context. I'm not really sure here...will sleep on it

But the numpy array case is easy, so let's start with that 😇